### ISSUE TITLE
Crashes on adding an address to a reply list

### ISSUE THREAD

**Issue by avisitor** [thunderbird/thunderbird-android#2973]
Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

I believe this is the same as "App crashed by typing address in the address field when composing a message #1251" except that bug was not fixed for when you reply to a message.

### Expected behavior
Should auto-complete the address and not crash

### Actual behavior
Crashes on the second character typed

### Steps to reproduce
1. Tap the left double-arrow at the bottom while viewing a message (to reply or forward)
2. Tap Reply
3. Tap in the To field after any existing addresses
4. Type in the first characters of an address to send to

### Environment
K-9 Mail version: 5.208

Android version: 7.0
[logcat.txt](https://github.com/k9mail/k-9/files/1565095/logcat.txt)

Account type (IMAP, POP3, WebDAV/Exchange): IMAP

**Comment by philipwhiuk** [comment]
```
2-07 14:07:33.882 E/AndroidRuntime(10268): java.lang.RuntimeException: An error occurred while executing doInBackground()
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.os.AsyncTask$3.done(AsyncTask.java:318)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.lang.Thread.run(Thread.java:762)
12-07 14:07:33.882 E/AndroidRuntime(10268): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.text.util.Rfc822Tokenizer.tokenize(Rfc822Tokenizer.java:51)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.text.util.Rfc822Tokenizer.tokenize(Rfc822Tokenizer.java:173)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.mail.Address.<init>(Address.java:51)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.mail.Address.<init>(Address.java:42)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.view.RecipientSelectView$Recipient.<init>(RecipientSelectView.java:540)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.activity.compose.RecipientLoader.fillContactDataFromCursor(RecipientLoader.java:241)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.activity.compose.RecipientLoader.fillContactDataFromQuery(RecipientLoader.java:192)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:110)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at com.fsck.k9.activity.compose.RecipientLoader.loadInBackground(RecipientLoader.java:25)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:312)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:69)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:66)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at android.os.AsyncTask$2.call(AsyncTask.java:304)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12-07 14:07:33.882 E/AndroidRuntime(10268): 	... 3 more
```

We are trying to add recipients with no email.

**Comment by avisitor** [comment]
These recipients all have email - they are in my Google address book.

Was this fix included in 5.403? That version still crashes the same way.

**Comment by philipwhiuk** [comment]
No it wasn't added to the 5.4 branch.

@cketti is it worth cherry-picking?
